client/lib/site/timezone.js

```diff
@@ -20,12 +20,12 @@ import moment, { MomentInput, Moment } from 'moment-timezone';
  *                        Moment with gmtOffset applied if no timezone is provided.
  *                        If neither is provided, the original moment is returned.
  */
-export function applySiteOffset( input, { timezone, gmtOffset } ) {
+export function applySiteOffset( input, { timezone, gmtOffset, keepLocalTime = false } ) {
```

Comment:
`keepLocalTime` is `false` by default to avoid changing the behavior in other places of the app.